### ISSUE TITLE
Git: Support editing the commit message in a text editor

### ISSUE THREAD

**Issue by TrevorBurnham** [microsoft/vscode#30562]
When I make a commit, I enjoy writing the commit message in an editor window containing a complete diff of my staged changes. I can do that from the terminal by running the command:

```
git commit --verbose
```

However, VSCode doesn't seem to support that commit mode out of the box. Perhaps it could be the fallback behavior if you run one of the `Git: Commit` commands and enter a blank commit message. (Currently, that fails silently.) Or better yet, how about a `git.verboseCommit` setting that, set to `true`, would make the `Git: Commit` commands take me directly to the verbose commit editor?

**Comment by joaomoreno** [comment]
You expect to edit the commit message in a full blown text editor?

**Comment by TrevorBurnham** [comment]
Indeed. That's been my workflow for many years. I find it invaluable to see all my changes in one place as I decide how to describe what I changed. It also makes it easy for me to spot any stray `debugger` statements and the like before I finalize the commit.

**Comment by rmunn** [comment]
I'll repost my comment from https://github.com/Microsoft/vscode/issues/2718#issuecomment-289389110 where I asked for some specific features for the Git commit message editor (whether it's in a full editor window or not, though being in a full editor window would probably make it easier to implement this request):

... [T]he rule for Git commit message word wrapping should be very simple:

1. First line warns if it's more than 50 columns. (This is the current behavior).
1. Second line warns if it isn't blank.
1. Rest of the commit wraps on spaces (or on any whitespace, but I recommend sticking with plain 0x20 space characters only for simplicity) at a user-configurable column, defaulting to 72 since that's the current standard recommended by Git.
1. Any line in the commit that does NOT contain spaces does not wrap, but a warning pops up (similar to the warning if the first line is > 50 columns) saying "Your commit message contains # long line(s) which could not be wrapped because they do not contain spaces. If these lines are URLs, that's probably okay, but otherwise you should consider adding spaces so that the lines can be wrapped. Your settings are currently set to wrap any lines longer than 72 characters in your commit message." (With #, of course, being substituted for the actual number of too-long lines in the message.)
1. That warning can be disabled in the user or workspace config settings, for the sake of any teams who have standardized on languages like Thai which do not use spaces in the language. The word-wrapping behavior can also be completely disabled if the user so desires, but it starts configured to be ON.
1. (Optional). If the long line starts with the regex [a-zA-Z][-a-zA-Z+.]+: (see RFC 3986), it is assumed to be a URL and no warning is issued for that line, since it's usually not desirable to split a URL across two or more lines.

This would go a long way towards making VS Code's Git message editor usable. As of right now, I avoid using VS Code's Git integration because it doesn't format Git messages correctly according to the coding standards of just about every project I've worked on. So despite the fact that VS Code has pretty good Git integration for most things, I still just Alt-Tab to a terminal and write my Git commits in Vim, where I can set it up to do the right thing for commit messages (wrap at 72 columns, or at 70 for one project I work on that has slightly stricter rules than most).

**Comment by stkb** [comment]
@rmunn I'm not sure about bringing the issue of automatic wrapping up again in here in this issue that isn't really about it, though I see that you may have commented here because I'd closed #2718, which I've now re-opened.

Something I've learned from writing [stkb/Rewrap](https://github.com/stkb/rewrap) is that wrapping isn't as simple as you'd first think and people have all sorts of requirements (ok, like just about every project then).

My workflow currently is to go the source control pane and look through the file diffs, then if it's a short commit message just type it in the box there, but it it's longer run `git commit` from the built-in terminal, which brings up a new vscode window where I have rulers set up at 50 and 72 columns and can use Rewrap to wrap the contents and check how it looks before committing.

So I don't see the need to use Vim, you can set up vscode as your git editor and for me it works fine. It would be nice if there was a faster way to switch to a full editor for the commit message, but I'd only put it in the "nice to have" category rather than essential.

**Comment by rmunn** [comment]
Yes, I reposted it because it looked like #2718 had been prematurely closed, and I didn't want that feature request to get lost. Now that #2718 has been re-opened, that's the right place to track that feature.

**Comment by albireox** [comment]
Any news on whether this will be implemented. I really like VSCode's Git workflow but I like the commit message style from Atom. I think this could be implemented as an option on commit so that, instead of showing the one-liner text box, it shows a fully fleshed editor (ideally as a split panel) using `git commit --verbose`. One of the advantages is that all rulers and autocomplete extensions that you add to the editor would be available.

Currently I'm using the [run-in-terminal](https://github.com/kortina/run-in-terminal) extension with a shortcut tied to a custom script that reproduces the behaviour of VSCode's commit. It works but not always quickly and seems like an overkill.

**Comment by JasonMFry** [comment]
Would love to see this feature implemented.

**Comment by JeanMertz** [comment]
> Currently I'm using the [run-in-terminal](https://github.com/kortina/run-in-terminal) extension with a shortcut tied to a custom script that reproduces the behaviour of VSCode's commit. It works but not always quickly and seems like an overkill.

@albireox would you mind sharing the implementation of that script? Do you do something other than shelling out to `git commit --verbose`, and have `code` set as your default Git editor?

**Comment by albireox** [comment]
The "script" I link to run-in-terminal is a simple fish shell function such as

```fish
function commit -d "Commits staged or all files"
    set result (eval "git diff --name-only --cached")
    if test -n "$result"
        git commit -v
    else
        git commit -v -a
    end
end
```

It basically checks if something has been staged and if not it commits everything.

**Comment by ccfz** [comment]
hi, @joaomoreno from what you wrote in https://github.com/Microsoft/vscode/issues/43585 I would like to add that in our company we want to help juniors write good commit messages. It not only helps understand what commits were about but more importantly makes junior think about the work they have done and most often we find that they go back to edit the commit. For example by splitting them into two or actually rewriting the code because they notice a single responsibility issue. To help them with this process we use git's template feature. This is what our template looks like in case you are curious. I think it be great if the warning that are diplayed in the VSCode's source control side-bar at the moment could also be shown in the editor. The auto-wrapping is of course even better but just the warning would also be a great benefit.



```
# 50-character subject line
#
# 72-character wrapped longer description. This should answer:
#
# * Why was this change necessary?
# * How does it address the problem?
# * Are there any side effects?
#
# Include a link to the ticket, if any.
#
# Add co-authors if you worked on this code with others:
#
# Co-authored-by: Full Name <email@example.com>
# Co-authored-by: Full Name <email@example.com>
# Please enter the commit message for your changes. Lines starting
# with '#' will be ignored, and an empty message aborts the commit.
#
# On branch address_picker
# Changes to be committed:
#	modified:   src/components/AddressPicker.js
#
# Changes not staged for commit:
#	modified:   package.json
#	modified:   src/components/AddressOption.js
#
```

**Comment by drorata** [comment]
Funny enough, I sometimes end up going to the terminal and running `git commit` so I'd have a proper editor for my commit message. 

@joaomoreno can you update the community on the status of this?

@TrevorBurnham Thanks for the `--verbose` that's going to be helpful :)

**Comment by joaomoreno** [comment]
No status, it's in the Backlog.

**Comment by rossburton** [comment]
Glad I'm not the only person whose commit workflow with vscode is to use the UI to commit with a dummy message and then immediately hit the terminal and do 'git commit --amend' so I can write a message in the full editor and not the silly popup.

**Comment by rossburton** [comment]
Regarding the closed issue:

> Most people don't really want a full editor taking away workbench space. They just want to type a message and move on.

I'd argue that isn't the case. You try submitting a quick one-liner message to any large project...

**Comment by jgunthorpe** [comment]
We are looking at using VS code for the Linux kernel. Commit messages typically run 20-70 lines long and generally require complex editing tasks like text reflow, copy & paste, careful indenting and managing the git tags list at the end. The tiny commit dialog is totally inadaquate for this task, it is required to have a fully featured editor.

Here are some examples of what is expected:

https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=28f94a44298c99c0db85539874b62f21d94fcaa7
(requires wrapping paragraphs)

https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=cc52d9140aa920d8d61c7f6de3fff5fea6692ea9
(requires manging indending and cut&paste alot of text)

**Comment by rdnlsmith** [comment]
Now that [global tasks](https://code.visualstudio.com/updates/v1_42#_user-level-tasks) are supported, you can use something like this (at least on Linux). You can then [assign a keybinding](https://code.visualstudio.com/docs/editor/tasks#_binding-keyboard-shortcuts-to-tasks) if you want.

(If anyone knows the appropriate `command` to do this on Windows, that would be amazing! Of course, it's easier if you have VS Code as your default Git editor to begin with.)

```json
{
    "label": "git commit --verbose",
    "type": "shell",
    "command": "GIT_EDITOR=\"code --wait\" git commit --verbose",
    "presentation": {
        "echo": false,
        "reveal": "silent",
        "focus": false,
        "panel": "shared",
        "showReuseMessage": false,
        "clear": false
    },
    "problemMatcher": []
}
```

**Comment by sdegutis** [comment]
Repeating my specific request from #93117:

> I'd like to be able to write a few paragraphs under the main git commit message. I can do that with `git commit` on the command line, but not in VS Code.
>
> I'd also like to be able to see the diffs of my commit while committing, so a modal would not be ideal, and the sidebar is too thin to add the text area into.
>
> So I suggest adding a panel at the top or bottom of the window, perhaps in the same area the terminal emulator is. That might be fine, although ideally I'd prefer the text area at the top.

Just throwing out my 2¢ and my preferred workflow.

**Comment by JohnnyCrazy** [comment]
I would like to tackle this and wonder if you already have some thoughts about this @joaomoreno?

My thoughts:

### UX/UI

From the UI/UX perspective, I don't see much obstacles, just some decisions need to be made.

When a specific setting (`git.commitUseEditor`) is set to `true`, the message box in the repository pane is replaced by a `Commit` button. When this button, or rather one of the `git.commit` commands, is pressed, a file `COMMIT_EDITMSG` should open the current window. When this file is closed, the commit should finish/continue. Basically the same workflow as if you would use `GIT_EDITOR=code git commit`.

Because it was mentioned often, another setting `git.verboseCommit` controls if you have the verbose output in your `COMMIT_EDITMSG`

### Git Integration

So this is the tricky part IMO. I researched how other extensions/editors do it and made a small list of possible options: 

#### 1. Use `GIT_EDITOR=${VSCODE_PATH} git commit`

Not a lot of integration is required in this case, but it has one large drawback: A full vscode process instance is started. This takes some time and may be too slow for some people. I wouldn't recommend this path


#### 2. Use `GIT_EDITOR=false git commit`

I found this method used in an old version of [fuvitive-vim](https://github.com/tpope/vim-fugitive), specifically [here](https://vi.stackexchange.com/questions/2544/how-to-manage-fugitive-commit-with-a-git-pre-commit-hook?rq=1). If the command is run, the commit will fail but `COMMIT_EDITMSG` will be generated. We can then open this file and once the users closes it, we can supply it via `git commit --file .git/COMMIT_EDITMSG`. It may sound like a hack, but it would be fast and AFAIK all git hooks would still work.

#### 3. Use a temporary script as `GIT_EDITOR`

This in the new version of [fuvitive-vim](https://github.com/tpope/vim-fugitive). Basically a temporary `.sh` script is created and used as `GIT_EDITOR`. This script checks periodically if another file exists (`.exit` file), and if so, exits. So the workflow could be similar to this:
1. Create temporary `.sh/.bat` script in filesystem
2. Use it as `GIT_EDITOR`. `COMMIT_EDITMSG` will be created and can be opened by vscode
3. When the users closes `COMMIT_EDITMSG`, create `.exit` file
4. `.sh/.bat` detects `.exit` file and closes --> git commit succeeds.

I personally think this is a clean way. However it depends heavily on the filesystem, not sure if you like that.

#### 4. Use self-made `COMMIT_EDITMSG`

This approach is taken by [git-plus](https://github.com/akonwi/git-plus) for atom. It basically trys to recreate a `COMMIT_EDITMSG` by calling various git functions and dumping the contents to a file. So it never uses the real `COMMIT_EDITMSG`, only a custom one. Not a fan of this solution. I think the users would like to edit the real `COMMIT_EDITMSG` with its default contents.

What are your thoughts? Any better ideas I'm not aware of?

I already looked at the code part of the git integration, which is already abstracted heavily. If we have the overall direction, I will investigate in which parts of the code I would add the changes :)

**Comment by segevfiner** [comment]
Atom has nice UI for this that allows you to just switch between editing in the commit message text box and the full editor rather than a setting that sets you to only use on or the other.

**Comment by JohnnyCrazy** [comment]
> 
> 
> Atom has nice UI for this that allows you to just switch between editing in the commit message text box and the full editor rather than a setting that sets you to only use on or the other.

Ah, seems like you're referring to the GitHub package. For future reference, they also use a custom file called `ATOM_COMMIT_EDITMSG`. It's empty when opened and has no info about the commit.

![gif](https://i.johnnycrazy.de/n8SiTPjf4vAWUn2agMhvD)

**Comment by JohnnyCrazy** [comment]
An update from my side:

I thought more about the whole git <-> vscode communication and found out, that there is an [existing IPC workflow used in the git extension](https://github.com/microsoft/vscode/tree/master/extensions/git/src/ipc) to ask for the password when cloning/pulling repos. Knowing this, I made a small Proof of Concept based on the following principles:

* Introduced `git.useEditorToCommit: boolean`, which specifies if the editor should be used for committing. Note, you can still use the small box on the repository panel, but when running any commit command or CTRl+ENTER while the commit inputbox is still empty, it will trigger a editor commit.
* Introduced `git.verboseCommit: boolean`, should be self-explanatory, adds verbose output to `COMIT_EDITMSG`
* When an editor commit is triggered, the following things happen:
	* No message is supplied to `git commit`, but `GIT_EDITOR` points to a custom`.sh/.bat`, which starts a nodejs process and connects via IPC & HTTP to the main process. This call also contains the `COMMITEDIT_MSG` path.
	* The main process gets the path and opens it as an editor. The HTTP-Server does not immediately send a response. It waits until the file is closed. This has the effect, that git commit waits until the user is finished with editing the message

![img](https://i.johnnycrazy.de/8q8zY7zOQMcVuwGivan5X)
(In this example I press CTRL+ENTER on the inputbox)

Problems:
* Since the HTTP-Call is kept open and no data is sent, it timeouts after some time. This triggers and empty commit message and aborts the call.

Before going further into the rabbit hole, I'll wait for feedback from @joaomoreno. Maybe that's not at all what you have imagined :P

**Comment by JohnnyCrazy** [comment]
Small bump for feedback from @joaomoreno and a possible fix for the problem stated above:

The default timeout of `http.Server`, which is used by the IPC, is 2 minutes. A simple solution would be disabling it. This is, btw, [also the default in node 13](https://github.com/nodejs/node/pull/27558).

**Comment by joaomoreno** [comment]
@JohnnyCrazy This is great, exactly what we should do. Feel free to start a PR.

**Comment by jedsmith** [comment]
Excited to see this merge. This is the main dealbreaker to a VS Code workflow for Git for me.

Nice work, @JohnnyCrazy.

**Comment by CyanBook** [comment]
Does someone have an update on this?

**Comment by ytimenkov** [comment]
I wonder would it be possible to make a small button on the right of commit message to pop it up?
From user perspective it's good for small messages you can type directly, for large messages you can edit in an editor and then sync it back.

VS Code team has done a lot of improvements to that edit field 👍 the last inconvenience is dynamic word wrap: even though 50/80 character lines are warned the reflowed text looks strange when only one or two words are wrapped. Not sure if scrollbar here is a good option.

**Comment by ionut-botizan** [comment]
Seeing how this is still not done after almost 4 years since the issue has been created, I'd like to share my perspective on this:

* I would have started by just adding a button in the commit dialog and implementing a VSCode command like _"Edit the commit message using text editor"_. The command could be triggered by the aforementioned button, by using the command palette or by using a keyboard shortcut previously assigned to it. This command would just run `git commit` in a shell and rely on Git's `core.editor` configuration to open the user defined editor. This would have been more than enough for advanced users. They would've have configured VSCode as the `core.editor` and the `.git/COMMIT_EDITMSG` file would've have opened in VSCode.
* On future releases, I would improve the experience for the rest of the users by detecting whether VSCode was already set or not as the Git `core.editor` by running first `git config core.editor` in a shell and see if the executable points to VSCode. If not, I would first notify users about that and ask them whether they want to continue or not. If they chose "Yes", then proceed as before with running `git commit`
* The next releases would take things further and, if, after detection, it appears that `core.editor` is set to something other than VSCode, VSCode would offer the option to set itself as the default Git `core.editor`, maybe giving the option to do it globally, with `git config --global` or just for the current project, using `git config --local`

For me (and I'm willing to bet that for many others) that would have been more than enough. Basically that's what I'm doing right now: use VSCode to easily identify and inspect the changes and stage the files that I need (or just parts of some files) and then go to the terminal and run `git commit` which opens a VSCode editor in a tab where I edit the commit message.

This whole process works. You should not try to re-invent it. You should just automate what we're already doing manually by piggybacking on existing Git functionality.

Anyways, these are my 2¢. 🙂

**Comment by calebmeyer** [comment]
I've bounced around a bunch of different editors/IDEs since starting my professional programming career. I started on notepad++. When an intern showed me Sublime Text, I switched to it, and learned vim from its vintage mode.

**Comment by calebmeyer** [comment]
TL;DR: Do what https://atom.io/packages/git-plus does for commits
![image](https://user-images.githubusercontent.com/3342777/132376214-bff5b312-a65f-45a4-831a-d2be7c4d40d0.png)

I've switched around among a bunch of different editors/IDEs since I started programming professionally.

I started on notepad++, and switched to Sublime Text when an intern told me about it. I learned vim through Sublime's vintage mode, and used it to switch to the actual vim for a while, which I still use for quick edits at the command line. I went from there to spacemacs, which had way more IDE like functionality, and showed me the importance of the `<leader>` key. Then atom came out, and it was revolutionary. Now I had an editor I could actually feel comfortable hacking on, and I did just that.

On topic: for git commits, all of {vim, emacs, atom} had the ability to use a single command to summon a full text editor which showed you what was staged and let you write a commit message. It had basic syntax highlighting, but most importantly it didn't break my flow. It was `space g c` (leader, git-commit) to summon it, and `esc f s` (normal mode, file-save) to commit. For short commits, it takes all of five seconds. Importantly, it doesn't require touching the mouse, and you get your cursor right back where you came from before the commit.

When I switched to VS Code recently, this [has been my biggest pain point](https://stackoverflow.com/questions/68986068/auto-close-git-panel-after-commit-in-visual-studio-code). I really want to be able to commit quickly and have it just disappear when I'm done. Syntax highlighting would be nice, and the approaches outlined above would satisfy my requirements.

**Comment by askhasni** [comment]
Is this implemented yet? One liner git commit messages are not helping anyone, it just adds an extra step for commit amend.

**Comment by 1Mark** [comment]
Any progress on this?

**Comment by rossburton** [comment]
🎉

**Comment by albireox** [comment]
This is not working for me in Insiders. If I enable 
```
"git.verboseCommit": true,
"git.useEditorAsCommitInput": true
```
and try to commit I get an error
```
Git: '/Applications/Visual Studio Code - Insiders.app/Contents/Resources/app/extensions/git/dist/scripts/git-editor.sh': /Applications/Visual Studio Code - Insiders.app/Contents/Resources/app/extensions/git/dist/scripts/git-editor.sh: No such file or directory
```
It seems there's a typo in the path. The correct one should be `/Applications/Visual Studio Code - Insiders.app/Contents/Resources/app/extensions/git/dist/gitEditor/scripts/`. However, if I try to create the `dist/scripts/` directory and symlink `git-editor.sh` from the correct location, I get this error
```
[2022-05-26T15:51:35.065Z] error: There was a problem with the editor ''/Applications/Visual Studio Code - Insiders.app/Contents/Resources/app/extensions/git/dist/scripts/git-editor.sh''.
Please supply the message using either -m or -F option.
```

**Comment by lszomoru** [comment]
Quick update on this. While we have ironed out all issues that were identified, we simply ran out of time for this change to be included in the `May 2022` milestone. The changes will be committed to the main branch as soon as it opens up for the `June 2022` milestone work. I will update this issue as soon as the changes are released in Insiders. Thank you very much for your patience.

**Comment by lszomoru** [comment]
Today's Insiders release (2022-06-10) contains the changes to use the editor for the git commit input. You can enable this capability using the `git.useEditorAsCommitInput`. After enabling the setting you will have to restart VS Code so that the `GIT_EDITOR` environment variable is set. Give it a try and let me know if you run into any issues. 

Thanks again to @JohnnyCrazy for his contribution as well as to all of you for your patience.

**Comment by happyteque** [comment]
Gosh here we go again... why are you people so obsessed with constantly changing defaults settings?? I have had vim for years and suddenly something changes and I have just wasted half an hour of my time trying to work out what to change to bring it back as it has been for years.

Pro TIP: unless you are fixing something that was breaking DON'T. CHANGE. DEFAULT. SETTINGS.

Thanks

**Comment by happyteque** [comment]
Additionally it's not synced, so now I have to manually go and change it on every machine I use... 🙄  ...

**Comment by segevfiner** [comment]
I'll just lueae this here:
![XKCD Workflow](https://imgs.xkcd.com/comics/workflow.png)

**Comment by joaomoreno** [comment]
> Pro TIP: unless you are fixing something that was breaking DON'T. CHANGE. DEFAULT. SETTINGS.

Maybe it isn't broken for you, but it is for users who are not comfortable with vim. Furthermore, being able to act as the default editor for git commit messages, we are now in the fortunate position to be able to provide additional value (think issue completion, author completion, semantic versioning linting, etc.).

> Additionally it's not synced, so now I have to manually go and change it on every machine I use... 🙄 

Feel free to enable [Settings Sync](https://code.visualstudio.com/docs/editor/settings-sync).

@happyteque One more thing: please think twice about arbitrarily offending people online. Just because there's hardware between you and another person, doesn't mean you can be impolite.

**Comment by AlekSi** [comment]
More xkcd

[![](https://imgs.xkcd.com/comics/internet_argument.png)](https://xkcd.com/438/)

**Comment by happyteque** [comment]
I understand why some people may like it, I am only objecting to the defaults changing for everyone, @joaomoreno. 
Git itself handled such changes better - when they enforce merge strategies, they stared showing a message telling people they now had a new option and had to make a choice, and told them what their options were. Whereas you guys just changed one of the settings without warning nor explanations. It's not the first time you do this, I still remember the train wreck whith the Terminal panels suddenly not being movable to the right anymore.

As for @segevfiner, he was the one out of line with his sarcastic, snappy reply. Perhaps you should have words with him, but I understand it's easier to gang up on ${WEIRD_STRANGER}.

**Comment by theAkito** [comment]
> If Vim is "horrifying" to you, I suggest a career switch to organic farming. What an arrogant smartarse

Thank you for appropriately assessing your own behaviour. This way, nobody has to tell you, how you sound.



> Every change breaks someone's workflow, which is why you should keep to a minimum unless you are solving a real problem. You weren't.

Obviously, there was a real issue, hence this issue. Additionally, if you are heating up your spacebar for years and think that's alright, then that does not mean nobody should break that behaviour. If you are doing it wrong for years, it's time, since years, to change your wrong behaviour.

All that you said in this issue is basically just a psychological defense mechanism, which automatically kicked and has only the purpose of making you look less like an arrogant dumbass. You are not being honest. You are not trying to find the truth or debate honestly about what is better. You just childishly came here -- initially caused by your wrongful assumption, that nothing will ever change and everything will ever stay the same with software, which evolves on a monthly basis -- and try to fart your fault out of your arse onto others, because you got used to heating up your spacebar for pressing the control button.

If you are not being able to adjust to improvements in your software tools, you should probably switch to a job, which is already fully optimised. However, do not switch to any type of farming, because there you would need to constantly adjust to new methods and technical possibilities, too. They use pretty high-tech machines there, nowadays.

> I am only objecting to the defaults changing for everyone

The same criticism you already read applies the same way to defaults. If a default is crap, it should be changed. Simple as that. Just because you are still stuck in the terminal in this specific situation, that does not mean everyone is and wants to be.

> Whereas you guys just changed one of the settings without warning nor explanations. It's not the first time you do this, I still remember the train wreck whith the Terminal panels suddenly not being movable to the right anymore.

Actually, I have experienced the same. Do you know what I do then? I read the most recent changelog, perhaps even CTRL+F for a keyword and there is my result. All clear now. Takes like two minutes in total.


> he was the one out of line with his sarcastic, snappy reply.

It wasn't sarcastic, though. You do the same as the guy heating up his spacebar. It's literally a description of what you were complaining about.


> I understand it's easier to gang up on ${WEIRD_STRANGER}.

Again, one of those psychologically caused fallacies for mindlessly defending oneself.
In case it makes you happier:

I am just another VS Code user who followed this issue for such a long time. Was waiting for a fix, as many others did. Now you came here with your smartarse attitude and made it seem like this was a bad change, just because you want to keep heating up your spacebar. So, I thought, I am going to tell you what's going on, because you are being simply unfair and absolutely dishonest.

**Comment by happyteque** [comment]
QED

I am sure everyone's really interested in reading @theAkito's vitriolic tome and his personal attacks. @joaomoreno will no doubt scold me for it. It's probably time to lock this ticket - I can only repeat the invitation to being more careful with handling default settings changes. As for the deranged comment above, I cannot help but note the @theAkito's mission statement is "Free yourself from yourself". I can see why they'd write that.

**Comment by joaomoreno** [comment]
Alright that's it, locking this down. Thanks for making everyone's day a little bit worse, @happyteque. I really hope you got something out of it, because nobody else did.
